core/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -393,10 +398,9 @@ private void internalRecoverFromStore(IndexShard indexShard) throws IndexShardRe
                     logger.debug("failed to list file details", e);
                 }
                 indexShard.performTranslogRecovery(indexShouldExists);
-                assert indexShard.shardRouting.primary() : "only primary shards can recover from store";
                 indexShard.getEngine().fillSeqNoGaps(indexShard.getPrimaryTerm());
+                indexShard.finalizeRecovery(indexShard.getLocalCheckpoint());
             }
-            indexShard.finalizeRecovery();
```

Comment:
why not just make this
```
indexShard.finalizeRecovery(indexShard.getLocalCheckpoint());
```
which applies to all store recoveries instead of moving the same call to the branches above?